test/ability_defeatist.c

```diff
@@ -0,0 +1,46 @@
+#include "global.h"
+#include "test_battle.h"
+
+ASSUMPTIONS
+{
+    ASSUME(gBattleMoves[MOVE_TACKLE].split == SPLIT_PHYSICAL);
+    ASSUME(gBattleMoves[MOVE_ECHOED_VOICE].split == SPLIT_SPECIAL);
+}
+
+SINGLE_BATTLE_TEST("Defeatist halves Attack when HP <= 50%", s16 damage)
+{
+    u32 hp;
+    PARAMETRIZE { hp = 400; }
+    PARAMETRIZE { hp = 200; }
+    GIVEN {
+        PLAYER(SPECIES_ARCHEN) { Ability(ABILITY_DEFEATIST); HP(hp), MaxHP(400);}
+        OPPONENT(SPECIES_WOBBUFFET);
+    } WHEN {
+        TURN { MOVE(player, MOVE_TACKLE); MOVE(opponent, MOVE_CELEBRATE); }
+    } SCENE {
+        ANIMATION(ANIM_TYPE_MOVE, MOVE_TACKLE, player);
+        HP_BAR(opponent, captureDamage: &results[i].damage);
+        ANIMATION(ANIM_TYPE_MOVE, MOVE_CELEBRATE, opponent);
+    } FINALLY {
+        EXPECT_LT(results[1].damage, results[0].damage);
```

Comment:
Shouldn't this be half instead of "less than"?

Reply:
The Attack and Special Attack stats are halved, not the actual damage output itself as far as I can tell

Reply:
I would expect that halving the stats would roughly correspond to halving the damage. Does `EXPECT_MUL_EQ(results[0].damage, Q_4_12(0.5), results[1].damage);` fail?

(`EXPECT_MUL_EQ` has a little bit of tolerance to account for rounding errors, so it's not quite the same as `EXPECT_EQ(results[0].damage / 2, results[1].damage);`)